integration/helpers.go

```diff
@@ -184,3 +184,22 @@ func CollectKeys(t testing.TB, doc bson.D) []string {
 
 	return res
 }
+
+// AssertEqualWriteError compares expected mongo.WriteError Message and Code with actual error.
+func AssertEqualWriteError(t *testing.T, expected *mongo.WriteError, actual error) bool {
```

Comment:
I'm not sure, but it also can shadow the MongoDB errors checking for example when we expect in tests a certain error type, but MongoDB gives another type, it will bypass.

Reply:
No shadowing here - it can be seen in deep debug on mongo driver.